src/NexusMods.DataModel/ApplyService.cs

```diff
@@ -32,29 +40,104 @@ public async Task<Loadout> Apply(LoadoutId loadoutId)
+        );
+
         try
         {
             await loadout.Apply();
         }
         catch (NeedsIngestException)
         {
             _logger.LogInformation("Ingesting loadout {LoadoutId} from {GameName} {GameVersion}", loadout.LoadoutId,
-                loadout.Installation.Game.Name, loadout.Installation.Version);
-            await loadout.Ingest();
-            _logger.LogInformation("Applying loadout {LoadoutId} to {GameName} {GameVersion}", loadout.LoadoutId,
-                loadout.Installation.Game.Name, loadout.Installation.Version);
-            await loadout.Apply();
+                loadout.Installation.Game.Name, loadout.Installation.Version
+            );
+
+            Loadout lastAppliedLoadout;
+            var lastAppliedRevision = GetLastAppliedLoadout(loadout.Installation);
+            if (lastAppliedRevision is not null)
+            {
+                var lastLoadout = _loadoutRegistry.GetLoadout(lastAppliedRevision);
+                lastAppliedLoadout = lastLoadout ?? throw new Exception("Loadout not found for last applied revision");
+            }
+            else
+            {
+                // There is apparently no last applied revision, so we'll just use the loadout we're trying to apply
+                lastAppliedLoadout = loadout;
+            }
+
+            var loadoutWithIngest = await lastAppliedLoadout.Ingest();
+
+            // Rebase unapplied changes on top of ingested changes
+            var mergedLoadout = _loadoutRegistry.Alter(loadout.LoadoutId, $"Rebase unapplied changes on top of ingested changes in loadout: {loadout.Name}",
+                l => l.Installation.GetGame().Synchronizer.MergeLoadouts(loadoutWithIngest, loadout)
+            );
+
+            _logger.LogInformation("Applying loadout {LoadoutId} to {GameName} {GameVersion}", mergedLoadout.LoadoutId,
+                mergedLoadout.Installation.Game.Name, mergedLoadout.Installation.Version
+            );
+
+            await mergedLoadout.Apply();
+            return mergedLoadout;
         }
 
         return loadout;
     }
 
+    public async Task<Loadout> Ingest(GameInstallation gameInstallation)
+    {
+        var lastAppliedRevision = GetLastAppliedLoadout(gameInstallation);
+        if (lastAppliedRevision is null)
+        {
+            throw new InvalidOperationException("Game installation does not have a last applied loadout to ingest into");
+        }
+
+        var lastLoadout = _loadoutRegistry.GetLoadout(lastAppliedRevision);
+        var lastAppliedLoadout = lastLoadout ?? throw new Exception("Loadout not found for last applied revision");
+        
+        var loadoutWithIngest = await lastAppliedLoadout.Ingest();
+        
+        // Get the latest loadout revision
+        var latestRevision = _loadoutRegistry.Get(loadoutWithIngest.LoadoutId);
+        if (latestRevision is null)
+            throw new Exception("No latest revision found for last applied loadout");
```

Comment:
Don't throw an exception of type `Exception`. Try to find an exception type that has more meaning or create a custom `Exception` class.